src/util/log-express-error-response.ts

```diff
@@ -0,0 +1,14 @@
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
+import Logger from "bunyan";
+
+export const logExpressErrorResponse = (logger?: Logger) =>
+	(req: Request, res: Response, next: NextFunction) => {
+		res.once("finish", async () => {
+			// if status is under 200 or higher/equal to 400, but not 503 in maintenance mode, log it to figure out issues
+			if ((res.statusCode < 200 || res.statusCode >= 400) && !(res.statusCode === 503 && await booleanFlag(BooleanFlags.MAINTENANCE_MODE, false))) {
+				(logger || req.log).warn({ res, req }, `Returning HTTP response of '${res.statusCode}' for path '${req.path}'`);
```

Comment:
Maybe it should be opposite `(req.log || logger)` so if there is no logger in the request we use the default one

Reply:
logger is being injected in the function.  we should use the one made explicit first, not the default.